### PR TITLE
[#151190515] Order screen - fix dummy 700

### PR DESCRIPTION
**Pull request by guillaumedebavelaere**

### CODE REVIEW

File: TradeItIosTicketSDK2/TradeItOrdersViewController.swift [hunk 1]
```diff
@@ -86,8 +88,6 @@ class TradeItOrdersViewController: TradeItViewController, TradeItOrdersTableDele
                 withLinkedBroker: self.linkedBrokerAccount?.linkedBroker,
                 onViewController: self
             )
```
Comment (guillaumedebavelaere) [inline TradeItIosTicketSDK2/TradeItOrdersViewController.swift]: There is an issue with dummy 600 though. We will get an alert with session expired - cancel - refresh. With the refresh, the screen won't be updated.
If we add the refresh in an onFinished, this will loop in case of error:
```
self.alertManager.showAlertWithAction(
                error: error,
                withLinkedBroker: self.linkedBrokerAccount?.linkedBroker,
                onViewController: self,
                onFinished: {
                    self.refreshRequested(onRefreshComplete: onRefreshComplete)
                }
            )
```
Thoughts ?

Comment (guillaumedebavelaere) [reply TradeItIosTicketSDK2/TradeItOrdersViewController.swift]: Maybe we should call showAlertWithAction only is it is a 700 error ?

**Comment by bluk16** [comment]
The alert is changed. When I select `refresh`, it is not redirected to the OAuth login screen.

screen shot 2017-09-19 at 9 32 30 am

**Comment by guillaumedebavelaere** [comment]
This is the 600 error. As I mentioned, the refresh doesn't refresh the screen. You will need to pull to refresh to trigger the 700 error.

**Comment by bluk16** [comment]
Thanks. I see what @gdebavelaere said here now.

File: TradeItIosTicketSDK2/TradeItOrdersViewController.swift (hunk 1 above)
Comment (jsom) [reply TradeItIosTicketSDK2/TradeItOrdersViewController.swift]: Is `onFinished` called if a user clicks Cancel?

Comment (guillaumedebavelaere) [reply TradeItIosTicketSDK2/TradeItOrdersViewController.swift]: @jsom Yes it is

**Comment by jsom** [comment]
@guillaumedebavelaere I think we leave this as is for now. I believe it's behaving the same way that the rest of the SDK does. I don't want to change `showAlertWithAction` until we have decent test coverage on it because I remember we did it in the past and it led to other edge cases breaking.

**Comment by guillaumedebavelaere** [comment]
Ok, good to merge then ? @bluk16 the dummy 700 error is ok ?
